File: Top_Interview_150/Array_String/python/45.py
from typing import List


class Solution:
    def jump(self, nums: List[int]) -> int:
        count = last = max_index = 0

        for i in range(len(nums) - 1):
            max_index = max(max_index, i + nums[i])

            if i == last:
                last = max_index
                count += 1

        return count

File: Top_Interview_150/Array_String/cpp/45.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int jump(vector<int> &nums)
    {
        int count = 0;
        int last = 0;
        int max_index = 0;

        for (int i = 0; i < nums.size() - 1; ++i)
        {
            max_index = max(max_index, i + nums[i]);

            if (i == last)
            {
                last = max_index;
                count++;
            }
        }

        return count;
    }
};